applications/tari_base_node/src/main.rs

```diff
@@ -182,15 +185,31 @@ fn setup_runtime(config: &GlobalConfig) -> Result<Runtime, String> {
         .map_err(|e| format!("There was an error while building the node runtime. {}", e.to_string()))
 }
 
-/// Set the interrupt flag on the node when Ctrl-C is entered
-fn handle_ctrl_c(rt: &Runtime, flag: Arc<AtomicBool>) -> task::JoinHandle<Result<(), String>> {
-    rt.spawn(async move {
-        signal::ctrl_c().await.map_err(|e| e.to_string())?;
-        info!(
-            target: LOG_TARGET,
-            "Termination signal received from user. Shutting node down."
-        );
-        flag.store(true, Ordering::SeqCst);
-        Ok(())
-    })
+async fn cli_loop(shutdown_flag: Arc<AtomicBool>) {
```

Comment:
This is bad because this is not really async. While this is waiting for input it is taking up a whole thread that cannot be released to do other work. 

I suggest looking at async alternatives - 
https://docs.rs/async-std/0.99.9/async_std/io/fn.stdin.html
https://docs.rs/futures/0.3.1/futures/io/trait.AsyncBufReadExt.html#method.read_line

Reply:
I see rustyline adds a few more (pretty cool) features than simply reading a line. Think you should just run this loop on the main thread and use message passing and the `tari_shutdown` crate to enable the console commands to perform actions in the async code.

Reply:
I think we just have to accept this. Most runtimes (Android, Node etc.) have a "UI" thread that basically sits waiting for input and responding to it. 

The decision is: use thread:spawn and figure out how to communicate with other subsystems. Or just let tokio manage the threading and make sure we have ample  core threads

Reply:
Might be missing what needs to happen here but I don't understand what benefit there is to making this an async function. It could take the runtime and spawn/block on whatever it needs to when commands are issued.

Reply:
This will not be just used to issue a shutdown command. Thats only a sub part of it. It will be used for getting UI feedback on the basenode and wallet api. For example getting the difficulty, or sending your mined tari some where.